motionplan/motionPlanner.go

```diff
@@ -28,8 +26,11 @@ type motionPlanner interface {
 	smoothPath(context.Context, []node) []node
 	checkPath([]frame.Input, []frame.Input) bool
 	checkInputs([]frame.Input) bool
-	getSolutions(context.Context, spatialmath.Pose, []frame.Input) ([]*costNode, error)
-	opt() *plannerOptions
+}
+
+type parallelMotionPlanner interface {
```

Comment:
We shouldn't be pre-emptively doing this, as we don't know what parallel non-RRT algorithms will look like. Let's stick with `motionPlanner` and `rrtParallelPlanner` rather than `rrtMotionPlanner` and `parallelMotionPlanner`. We can reorganize those if the need comes up.

Reply:
I'm good to rename back to rrtParallelPlanner, but I don't want to reorganize back to just `rrtMotionPlanner`.  As it currently stands I've abstracted out all the rrt specific code to the rrt.go function and I would like to keep it this way as it makes it easier to build on, regardless of if we have another type of algorithm